tests/Unit/Router/RouterMatchTest.php

```
@@ -0,0 +1,175 @@
+        });
+    }
+
+    public function anyHttpMethodProvider(): Generator
+    {
+        yield ['METHOD_GET' => Request::METHOD_GET];
+        yield ['METHOD_CONNECT' => Request::METHOD_CONNECT];
+        yield ['METHOD_DELETE' => Request::METHOD_DELETE];
+        yield ['METHOD_HEAD' => Request::METHOD_HEAD];
+        yield ['METHOD_OPTIONS' => Request::METHOD_OPTIONS];
+        yield ['METHOD_PATCH' => Request::METHOD_PATCH];
+        yield ['METHOD_POST' => Request::METHOD_POST];
+        yield ['METHOD_PUT' => Request::METHOD_PUT];
+        yield ['METHOD_TRACE' => Request::METHOD_TRACE];
+    }
+
+    /**
+     * @dataProvider matchesMethodsProvider
+     */
+    public function test_match_matches_all_its_methods(string $testMethod, array $givenMethods): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/expected/uri';
+        $_SERVER['REQUEST_METHOD'] = $testMethod;
+
+        $this->expectOutputString('Expected!');
+
+        Router::configure(static function (Routes $routes) use ($givenMethods): void {
+            $routes->match($givenMethods, 'expected/uri', FakeController::class, 'basicAction');
+        });
+    }
+
+    public function matchesMethodsProvider(): Generator
+    {
+        yield [Request::METHOD_GET, [Request::METHOD_GET, Request::METHOD_POST]];
+        yield [Request::METHOD_POST, [Request::METHOD_GET, Request::METHOD_POST]];
+        yield [Request::METHOD_PATCH, [Request::METHOD_PATCH, Request::METHOD_PUT]];
+        yield [Request::METHOD_PUT, [Request::METHOD_PATCH, Request::METHOD_PUT]];
+    }
+
+    /**
+     * @dataProvider notMatchesMethodsProvider
+     */
+    public function test_match_does_not_matches_other_methods(string $testMethod, array $givenMethods): void
+    {
+        $_SERVER['REQUEST_URI'] = 'https://example.org/expected/uri';
+        $_SERVER['REQUEST_METHOD'] = $testMethod;
+
+        $this->expectOutputString('');
```

Should we throw an exception with a `404 error code`? @Tito-Kati @Chemaclass

---

Yeah! We currently don't have support for 404, so this is a nice follow up feature to implement ;)